test/ng/compileSpec.js

```diff
@@ -12111,6 +12141,626 @@ describe('$compile', function() {
+      expect(element.prop('asdf')).toBe(123);
+      expect(element.attr('asdf')).toBe('foo');
+    }));
+
+    it('should use the full ng-prop-* attribute name in $attr mappings', function() {
+      var attrs;
+      module(function() {
+        directive('attrExposer', valueFn({
+          link: function($scope, $element, $attrs) {
+            attrs = $attrs;
+          }
+        }));
+      });
+      inject(function($compile, $rootScope) {
+        $compile('<div attr-exposer ng-prop-title="12" ng-prop-super-title="34" ng-prop-my-camel_title="56">')($rootScope);
+        $rootScope.$apply();
+
+        expect(attrs.title).toBeUndefined();
+        expect(attrs.$attr.title).toBeUndefined();
+        expect(attrs.ngPropTitle).toBe('12');
+        expect(attrs.$attr.ngPropTitle).toBe('ng-prop-title');
+
+        expect(attrs.superTitle).toBeUndefined();
+        expect(attrs.$attr.superTitle).toBeUndefined();
+        expect(attrs.ngPropSuperTitle).toBe('34');
+        expect(attrs.$attr.ngPropSuperTitle).toBe('ng-prop-super-title');
+
+        expect(attrs.myCamelTitle).toBeUndefined();
+        expect(attrs.$attr.myCamelTitle).toBeUndefined();
+        expect(attrs.ngPropMyCamelTitle).toBe('56');
+        expect(attrs.$attr.ngPropMyCamelTitle).toBe('ng-prop-my-camel_title');
+      });
+    });
+
+    it('should not conflict with (ng-attr-)attribute mappings of the same name', function() {
+      var attrs;
+      module(function() {
+        directive('attrExposer', valueFn({
+          link: function($scope, $element, $attrs) {
+            attrs = $attrs;
+          }
+        }));
+      });
+      inject(function($compile, $rootScope) {
+        $compile('<div attr-exposer ng-prop-title="42" ng-attr-title="foo" title="bar">')($rootScope);
+        $rootScope.$apply();
+        expect(attrs.title).toBe('foo');
+        expect(attrs.$attr.title).toBe('title');
+        expect(attrs.$attr.ngPropTitle).toBe("ng-prop-title");
+      });
+    });
+
+    it('should disallow property binding on onclick', inject(function($compile, $rootScope) {
+      // All event prop bindings are disallowed.
+      $rootScope.onClickJs = function() {};
+      expect(function() {
+          $compile('<button ng-prop-onclick="onClickJs"></script>');
+        }).toThrowMinErr(
+          '$compile', 'nodomevents', 'Property bindings for HTML DOM event properties are disallowed.  ' +
+          'Please use the ng- versions (such as ng-click or ng-on-click instead of ng-prop-onclick) instead.');
+      expect(function() {
+          $compile('<button ng-prop-ONCLICK="onClickJs"></script>');
+        }).toThrowMinErr(
+          '$compile', 'nodomevents', 'Property bindings for HTML DOM event properties are disallowed.  ' +
+          'Please use the ng- versions (such as ng-click or ng-on-click instead of ng-prop-onclick) instead.');
+    }));
+
+    it('should process property binding in pre-linking phase at priority 100', function() {
```

Comment:
This description was copied from an `ng-attr-*` test, but doesn't describe the logic of `ng-prop-*` because we only assign the prop in the `$watch`er. This is unlike `ng-attr-*` that has special logic to assign the attribute asap. See discussion: https://github.com/angular/angular.js/pull/16614#discussion_r198717403

Reply:
For now I'll just update the description of this test.